sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java

```diff
@@ -42,8 +44,9 @@
      */
     @Override
     public Collection<Resource> getPersistentVolumesToBeDestroyed(Offer offer) {
-        List<Resource> resources = offer.getResourcesList().stream().
-                filter(resource -> ResourceUtils.getPersistenceId(resource) != null).collect(Collectors.toList());
+        List<Resource> resources = offer.getResourcesList().stream()
+                .filter(resource -> resource.hasDisk() && resource.getDisk().hasPersistence())
```

Comment:
Resource is a mesos protobuf. We should be able to resolve this once we've got our own classes for these types.